Dockerfile

```diff
@@ -0,0 +1,10 @@
+FROM python:3.8-slim-buster
+
+WORKDIR /flask_app
+
+COPY requirements.txt requirements.txt
+RUN pip3 install -r requirements.txt
+
+COPY . .
```

Comment:
If you want to copy the directory into the docker container, it may be a good idea. However when doing the development you'd need to re-build the image each time you make changes. Unless you have that directory re-mapped somewhere in docker-compose.yml.

Reply:
Btw, here's an example of a docker-compose.yml.

```---
version: '3'
services:
  postgres:
    image: postgres:10-alpine
    container_name: postgres
    environment:
      POSTGRES_PASSWORD: postgres
      POSTGRES_USER: postgres
    ports:
      - "5432:5432"
    expose: [5432]
    volumes:
      - postgres-data:/var/lib/postgresql/data

  redis:
    image: redis:6-alpine
    container_name: redis
    ports:
      - "6379:6379"
    expose: [6379]

  api:
    build:
      context: .
      dockerfile: Dockerfile.dev
    depends_on:
      - redis
      - elasticsearch
      - postgres
    ports:
      - "3000:3000"
      - "1234:1234"
    expose: [3000, 1234]
    environment:
      PGHOST: postgres
      PGUSER: postgres
      PGPASSWORD: postgres
      REDIS_URL: redis://redis:6379/1
    profiles:
      - api
    volumes:
      - api-data:/usr/local

volumes:
  elasticsearch-data: null
  postgres-data: null
  api-data: null
```

Reply:
It uses profile, so may not be something necessary for you right now.